src/datasets/vision/imagenet.jl

```diff
@@ -0,0 +1,186 @@
+    Defaults to `ImageNetReader.default_preprocess`, which applies a center-crop
+    and normalization using coefficients from PyTorch's vision models.
+- `inverse_preprocess`: inverse function of `preprocess` used in `convert2image`.
+    Defaults to `ImageNetReader.default_inverse_preprocess`.
+
+**Note:** When providing custom preprocessing functions, make sure both match.
+
+# Fields
+
+- `metadata`: A dictionary containing additional information on the dataset.
+- `split`: Symbol indicating the selected data partition
+- `dataset`: A `FileDataset` containing paths to ImageNet images as well as a `loadfn`
+    used to load images, which applies `preprocess`.
+- `targets`: An array storing the targets for supervised learning.
+- `inverse_preprocess`: inverse function of `preprocess` used in `convert2image`.
+
+# Methods
+
+$METHODS_SUPERVISED_ARRAY
+- [`convert2image`](@ref) converts features to `RGB` images.
+
+# Examples
+
+```julia-repl
+julia> using MLDatasets: ImageNet
+
+julia> dataset = ImageNet(:val);
+
+julia> dataset[1:5].targets
+5-element Vector{Int64}:
+ 1
+ 1
+ 1
+ 1
+ 1
+
+julia> X, y = dataset[1:5];
+
+julia> size(X)
+(5,)
+
+julia> size(X[1])
+(224, 224, 3)
+
+julia> X, y = dataset[2000];
+
+julia> convert2image(dataset, X)
+
+julia> dataset.metadata
+Dict{String, Any} with 8 entries:
+  "features_dir"      => "/Users/funks/.julia/datadeps/ImageNet/val"
+  "class_WNIDs"       => ["n01440764", "n01443537", "n01484850", "n01491361", "n01494475", "
+  "class_description" => ["freshwater dace-like game fish of Europe and western Asia noted f
+  "n_observations"    => 50000
+  "class_names"       => Vector{SubString{String}}[["tench", "Tinca tinca"], ["goldfish", "C
+  "metadata_path"     => "/Users/funks/.julia/datadeps/ImageNet/devkit/data/meta.mat"
+  "n_classes"         => 1000
+  "img_size"          => (224, 224)
+  "wnid_to_label"     => Dict("n07693725"=>932, "n03775546"=>660, "n01689811"=>45, "n0210087
+
+julia> dataset.metadata["class_names"][y]
+  3-element Vector{SubString{String}}:
+   "common iguana"
+   "iguana"
+   "Iguana iguana"
+```
+
+# References
+
+[1]: [Russakovsky et al., ImageNet Large Scale Visual Recognition Challenge](https://arxiv.org/abs/1409.0575)
+"""
+struct ImageNet <: SupervisedDataset
+    metadata::Dict{String,Any}
+    split::Symbol
+    dataset::FileDataset
+    targets::Vector{Int}
+    inverse_preprocess::Function
+end
+
+ImageNet(; split=:train, Tx=Float32, kws...) = ImageNet(Tx, split; kws...)
+ImageNet(split::Symbol; kws...) = ImageNet(; split, kws...)
+ImageNet(Tx::Type; kws...) = ImageNet(; Tx, kws...)
+
+function ImageNet(
+    Tx::Type,
+    split::Symbol;
+    img_size::Tuple{Int,Int}=(224, 224),
+    preprocess=ImageNetReader.default_preprocess,
+    inverse_preprocess=ImageNetReader.default_inverse_preprocess,
+    dir=nothing,
+    train_dir="train",
+    val_dir="val",
+    test_dir="test",
+    devkit_dir="devkit",
+)
```

Comment:
```suggestion
        Tx::Type,
        split::Symbol;
        img_size::Tuple{Int,Int}=(224, 224),
        preprocess=ImageNetReader.default_preprocess,
        inverse_preprocess=ImageNetReader.default_inverse_preprocess,
        dir=nothing,
        train_dir="train",
        val_dir="val",
        test_dir="test",
        devkit_dir="devkit",
    )
```

Reply:
If you have a preferred code style, I can run JuliaFormatter on all new files.